File: Euler/10/1.lua
#!/usr/bin/lua5.3
-- 7 seconds

-- Returns summation of all prime numbers till n
function summation(n)
 local function f(n)
  for i=3,math.floor(math.sqrt(n)),2 do
   if n%i==0 then return false end
  end
  return true
 end

 local sum = 2
 for i=3,n,2 do
  if f(i) then sum = sum + i end
 end
 return sum
end

print( summation(2000000) )

File: Euler/10/1.c
// gcc -lm 7.c
// doesn't work because I AINT MESSING WITH TYPES

#include <stdio.h>
#include <math.h>

int f(int n) {
 int limit = sqrt ( n );
 for (int i=3;i <= limit;i+=2)
  if (n%i==0) return 0;
 return 1;
}

int summation(int n){
 double sum = 2;
 for (int i = 3; i<n; i+=2)
  if ( f(i) ) sum+=i;
//  f(i) && sum+=i
 return sum;
}

int main() {
 printf("%i\n",summation(2000000));
 return 0;
}
